httpx/dispatch/proxy_http.py

```diff
@@ -167,8 +167,9 @@ def build_auth_header(self, username: str, password: str) -> str:
                 response=proxy_response,
             )
         else:
-            proxy_response.on_close = None
-            await proxy_response.read()
+            # Hack to ingest the response, without closing it.
+            async for chunk in proxy_response._raw_stream:
+                pass
```

Comment:
It's a hack, yes 😕 But it seems there's no public interface on `Response` that allows us to do this. Why is it important to not close the response in the first place? We don't do anything with it afterwards. Would it close the connection?

Reply:
Actually this point in the code surprised me too. Yes I think we're making sure that we maintain the connection. However I *actually* think that this is highlighting an issue here. I *think* that we probably actually want an explicit "proxy being established state" on the HTTP Connection, that will play into the stuff I was starting to think about on https://github.com/encode/httpx/pull/624. It wouldn't at all surprise me if we have some buggy behavior aroung this, such as later maintaining proxy connections that ought to have become closed.